modules/fedimint-lnv2-tests/src/bin/tests.rs

```diff
@@ -260,7 +265,15 @@ async fn test_payments(dev_fed: &DevJitFed) -> anyhow::Result<()> {
         await_receive_claimed(&client, receive_op).await?;
     }
 
-    info!("Testing LDK Gateway can pay HOLD invoice...");
+    while client.balance().await? < 9000 * 1000 {
+        sleep_in_test(
+            format!("Waiting for the full balance to become available to the client"),
+            Duration::from_secs(1),
+        )
+        .await;
+    }
```

Comment:
You can use [`retry`](https://docs.fedimint.org/fedimint_core/util/fn.retry.html) here